packages/utils/src/saml/pre-prod-tester.ts

```diff
@@ -0,0 +1,309 @@
+    return results;
+  } catch (error) {
+    console.error("Error executing SQL query:", error);
+    sequelize.close();
+    throw error;
+  }
+}
+
+async function getDrUrl(id: string): Promise<string> {
+  const sqlDBCreds = getEnvVarOrFail("DB_CREDS");
+  const sequelize = initDbPool(sqlDBCreds);
+  const query = `
+    SELECT cde.url_dr
+    FROM cq_directory_entry cde
+    WHERE cde.id = :id;
+  `;
+  try {
+    const results = await sequelize.query<QueryResult>(query, {
+      replacements: { id },
+      type: QueryTypes.SELECT,
+    });
+    sequelize.close();
+    return results[0].url_dr;
+  } catch (error) {
+    console.error("Error executing SQL query:", error);
+    sequelize.close();
+    throw error;
+  }
+}
+
+async function DQIntegrationTest() {
+  let successCount = 0;
+  let failureCount = 0;
+  let runTimeErrorCount = 0;
+
+  const results = await queryDatabaseForDQs();
+  const promises = results.map(async result => {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const dqResult = (result as any).data as OutboundDocumentQueryResp;
+    if (!dqResult.cxId || !dqResult.patientId || !dqResult.externalGatewayPatient) {
+      console.log("Skipping: ", dqResult.id);
+      return undefined;
+    }
+    const dqRequest: OutboundDocumentQueryReq = {
+      id: dqResult.id,
+      cxId: dqResult.cxId,
+      gateway: dqResult.gateway,
+      timestamp: dqResult.timestamp,
+      patientId: dqResult.patientId,
+      samlAttributes: samlAtributes,
+      externalGatewayPatient: dqResult.externalGatewayPatient,
+    };
+    try {
+      const dqResponse = await queryDQ(dqRequest);
+      return { dqResult, dqResponse };
+    } catch (error) {
+      console.error("Runtime error:", error);
+      throw error;
+    }
+  });
+
+  const responses = await Promise.allSettled(promises);
+  responses.forEach(response => {
+    if (response.status === "fulfilled" && response.value) {
+      const { dqResult, dqResponse } = response.value;
+      const resultDocumentReferences = new Set(
+        dqResult?.documentReference?.map(doc => `${doc.repositoryUniqueId}-${doc.docUniqueId}`)
+      );
+      const responseDocumentReferences = new Set(
+        dqResponse?.documentReference?.map(doc => `${doc.repositoryUniqueId}-${doc.docUniqueId}`)
+      );
+
+      if (
+        (responseDocumentReferences.size > 0 || resultDocumentReferences.size > 0) &&
+        responseDocumentReferences.size >= resultDocumentReferences.size
+      ) {
+        successCount++;
+      } else {
+        failureCount++;
+      }
+    } else if (response.status === "rejected") {
+      runTimeErrorCount++;
+    }
+  });
+
+  if (runTimeErrorCount > 0) {
+    console.log(`TEST FAILED: ${runTimeErrorCount} run-time errors occurred`);
+  } else {
+    console.log("TEST PASSED");
+  }
+  console.log(`DQ Success Count: ${successCount}`);
+  console.log(`DQ Failure Count: ${failureCount}`);
+}
+
+async function DRIntegrationTest() {
```

Comment:
steps:
1. get dqs from db 
2. send dqs
3. take document reference from dq result
4. get dr url from db
5. send dr

The reason we have to do this is because documentUniqueIds change for a lot of endpoints and the old ids in the DB no longer work